services/user.py

```diff
@@ -0,0 +1,59 @@
+from db.models import User
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None
+) -> User:
+    user = User()
+    user.username = username
+    user.set_password(password)
+
+    if email:
+        user.email = email
+
+    if first_name:
+        user.first_name = first_name
+
+    if last_name:
+        user.last_name = last_name
+
+    user.save()
+    return user
+
+
+def get_user(user_id: int) -> User:
+    return User.objects.get(id=user_id)
+
+
+def update_user(
+    user_id: int,
+    username: str = None,
+    password: str = None,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None
+) -> None:
+    user = User.objects.get(id=user_id)
+
+    if username:
+        if User.objects.filter(username=username).exists():
+            return
```

Comment:
The current implementation simply returns if the username already exists. Consider raising an exception or providing feedback to the caller to handle this case more explicitly.